src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1696,22 +1696,16 @@ struct WindowFunctionNthValue final : public WindowFunction
 
         int64_t offset = (*current_block.input_columns[
                 workspace.argument_column_indices[1]])[
-            transform->current_row.row].get<Int64>() - 1;
+            transform->current_row.row].get<Int64>();
 
-        if (offset < 0)
+        if (offset > INT_MAX || offset <= 0)
```

Comment:
Why INT_MAX while we have int64?

Reply:
We still can have more than 2bn rows in a block.

Reply:
Yes, initially this is not my code, but I guess it was done as some sort of soft limit.
Maybe @akuzm or @ryzuo can comment on this?
In the mean time I will increase it.

Reply:
Also lagInFrame/leadInFrame has the same limits, after d9ad06bd6d3